pkg/mimir/mimir.go

```diff
@@ -818,19 +832,18 @@ func (t *Mimir) Run() error {
 	return err
 }
 
-func (t *Mimir) readyHandler(sm *services.Manager) http.HandlerFunc {
+func (t *Mimir) readyHandler(sm *services.Manager, ready *atomic.Bool) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if !sm.IsHealthy() {
+		if !ready.Load() || !sm.IsHealthy() {
 			var serviceNamesStates []string
 			for name, s := range t.ServiceMap {
 				if s.State() != services.Running {
 					serviceNamesStates = append(serviceNamesStates, fmt.Sprintf("%s: %s", name, s.State()))
 				}
 			}
 
-			level.Debug(util_log.Logger).Log("msg", "some services are not Running", "services", serviceNamesStates)
-
-			httpResponse := "Some services are not Running:\n" + strings.Join(serviceNamesStates, "\n")
+			level.Debug(util_log.Logger).Log("msg", "component is stopping or some services are not Running", "services", serviceNamesStates)
+			httpResponse := "Component is stopping or some services are not Running:\n" + strings.Join(serviceNamesStates, "\n")
```

Comment:
I would actually write two different responses / log messages based on `shutdownRequested`. We should exactly tell the user why it's not ready, which is either because a shutdown has been requested or some services are not ready.